lib/strings.nix

```
@@ -996,7 +996,7 @@ rec {
     if isStringLike x then
       let str = toString x; in
       substring 0 1 str == "/"
-      && dirOf str == storeDir
+      && (dirOf str == storeDir || builtins.match "^/[A-Za-z0-9]{52}" str != null)
```

It must not be 53 characters or more for CA.

```suggestion
      && (dirOf str == storeDir || builtins.match "^/[A-Za-z0-9]{52}($|/)" str != null)
```

---

Also check the string context?